account-kit/smart-contracts/src/ma-v2/account/nativeSMASigner.ts

```diff
@@ -78,24 +138,18 @@ export const nativeSMASigner = (
 
     // we apply the expected 1271 packing here since the account contract will expect it
     async signMessage({ message }: { message: SignableMessage }): Promise<Hex> {
```

Comment:
try this with a this this is assuming we have we can infer the type from references to params sent down or 

```ts
async signMessage(this: SigningMethods<'eth_signTypedData_v4'>, {message}: MessageTrait) {// do things}
```
or do an overloaded type definition based on the type
```ts

const signMessage: Type extends 'eth_signTypedData_v4' ? (value: MessageTrait) => RT : never
const  signMessage = async ({message}) {// do things};
```